src/modules/student/entities/student-initial.entity.ts

```diff
@@ -19,4 +29,14 @@ export class StudentInitial extends BaseEntity {
 
   @Column('simple-array')
   bonusProjectUrls: string[];
+
+  @OneToOne(() => StudentProfile, (profile) => profile.id)
+  @JoinColumn()
+  profile: Profile;
+
+  @OneToOne(() => User, (user) => user.id)
+  @JoinColumn()
```

Comment:
ja myślę, że tu wystarczy jednokierunkowa relacja one-to-one, więc OneToOne() w encji usera wystarczy.
Poza tym JoinColumn nawet w relacji dwukierunkowej one-to-one ustawia się tylko po jednej stronie
sam musiałem tu zajrzeć to od razu zalinkuję żebyś sam sprawdził:
https://typeorm.io/one-to-one-relations

Reply:
ok